src/core/lib/channel/connected_channel.cc

```diff
@@ -381,7 +380,7 @@ class ClientStream : public Orphanable {
           nullptr;
       batch_payload_.recv_initial_metadata.peer_string = nullptr;
       server_trailing_metadata_ =
-          GetContext<FragmentAllocator>()->MakeClientMetadata();
+          GetContext<Arena>()->MakePooled<ServerMetadata>(GetContext<Arena>());
```

Comment:
Should this be MakePooled<ClientMetadata> ?

Reply:
no... this is the server trailing metadata, so metadata from the server (which is what the type indicates)... the previous code was wrong!

(that said, totally the same underlying type right now which is why it wasn't caught)